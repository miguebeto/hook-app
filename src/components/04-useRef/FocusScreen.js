import React, { useRef } from "react";
import "../02-useEffect/effect.css";

export const FocusScreen = () => {

  //creamos una referencia
  const inputRef = useRef();

  //cuando le damos click al boton selecciona el input seleccionando el valor dentro del input (similar al focus)
  const handleClick = () => {
    // document.querySelector("input").select();
    inputRef.current.select();
    console.log(inputRef);
  };

  return (
    <div>
      <h1>Focus Screen</h1>
      <hr />
      {/* Le asignamos la referencia al input y con el boton activamos la funcion que ejecuta el select del texto  */}
      <input ref={ inputRef } className="form-control" placeholder="Su nombre" />
      <button className="btn btn-outline-primary mt-2" onClick={handleClick}>
        Focus
      </button>
    </div>
  );
};
