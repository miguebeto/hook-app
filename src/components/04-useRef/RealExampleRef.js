import React, { useState } from "react";

import "../02-useEffect/effect.css";
import { MultipleCustomHooks } from "../03-examples/MultipleCustomHooks";

export const RealExampleRef = () => {
  //creamos un estado que tendra de initial state false y cambiara a true cuando ejecutamos la funcion seteadora
  const [show, setShow] = useState(false);

  return (
    <div>
      <h1>RealExampleRef</h1>
      <hr />
      {/* Si show es true muestra el componente  */}
      {show && <MultipleCustomHooks />}
      {/* cambia el valor de true a false y viceversa al dar click  */}
      <button className="btn btn-success mt-2" onClick={() => setShow(!show)}>Show/Hide</button>
    </div>
  );
};
