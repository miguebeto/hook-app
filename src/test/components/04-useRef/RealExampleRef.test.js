import React from 'react';
import { shallow } from 'enzyme';
import { RealExampleRef }  from '../../../components/04-useRef/RealExampleRef';
// import { MultipleCustomHooks } from '../../../components/03-examples/MultipleCustomHooks';

describe('Pruebas en <RealExampleRef />', ()=> {
    //debe retornar el componente sin cambios en el render
    const wrapper = shallow(<RealExampleRef />);


    test('Debe mostrarse correctamente', ()=>{

        //tomamos la foto del componente como se muestra originalmente
        expect(wrapper).toMatchSnapshot();
        //comprobamos que el componente no se debe mostrar inicialmente
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(false);
        
    });
    
    test('Debe mostrarse el componente <MultipleCustomHook /> al dar click en el boton', ()=>{

        //similamos dar click en el boto
        wrapper.find('button').simulate('click');
        //comprobamos que el component se encuentra renderizado
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true);
    });
});