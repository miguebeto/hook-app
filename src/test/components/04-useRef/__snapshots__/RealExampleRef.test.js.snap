// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <RealExampleRef /> Debe mostrarse correctamente 1`] = `
<div>
  <h1>
    RealExampleRef
  </h1>
  <hr />
  <button
    className="btn btn-success mt-2"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;
